src/datachain/lib/dc/datachain.py

```diff
@@ -443,9 +443,17 @@ def listings(
         )
         return listings(*args, **kwargs)
 
+    def persist(self) -> "Self":
+        """Save to temporary dataset that will be removed after the process ends.
```

Comment:
Can we improve the docs a bit?

- Don't mention dataset here (it's not important)
- Make sure that it's clear what is the use case here (what is the optimization here)